File: Leetcode/881.py
class Solution:
    def numRescueBoats(self, people, limit):
        count = 0
        people.sort()
        left = 0
        right = len(people)-1
        while left<=right:
            if people[left]+people[right]<=limit:
                left+=1
                right-=1
                count+=1
            else:
                right-=1
                count+=1
        return count

obj = Solution()
ret = obj.numRescueBoats([6,3,5,6,2,3],6)
print(ret)

File: Leetcode/881.cpp
#include<vector>
#include<iostream>
using namespace std;
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        int count = 0;
        sort(people.begin(),people.end());
        int left = 0;
        int right = people.size()-1;
        while (left<=right){
            if(people[left]+people[right]<=limit){
                left++;
                right--;
                count++;
            }
            else{
                right--;
                count++;
            }
        }
        return count;
    }
};